### ISSUE TITLE
io.netty.testsuite.transport.socket.SocketSslEchoTest fails with a core dump

### ISSUE THREAD

**Issue by radai-rosenblatt** [netty/netty#5890]
```
testsuite>$ mvn -Dtest=SocketSslEchoTest test
...
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
[jetty-alpn-agent] Using: alpn-boot-8.1.3.v20150130.jar
Running io.netty.testsuite.transport.socket.SocketSslEchoTest
...
17:06:10.865 [main] DEBUG i.n.u.internal.NativeLibraryLoader - Successfully loaded the library: netty-tcnative
```
ends with this:
```
17:06:51.803 [testsuite-nio-worker-4-2] DEBUG io.netty.handler.ssl.SslHandler - [id: 0x8977b904, L:/0:0:0:0:0:0:0:1:51409 - R:/0:0:0:0:0:0:0:1:51695] HANDSHAKEN: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
17:06:51.803 [testsuite-nio-worker-4-2] DEBUG i.n.t.t.socket.SocketSslEchoTest - HANDSHAKEN:
	client { sent: 16384, rcvd: 0, nego: 0, cipher: SSL_NULL_WITH_NULL_NULL },
	server { rcvd: 0, nego: 1, cipher: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 }
17:06:51.803 [testsuite-oio-worker-6-1] DEBUG io.netty.handler.ssl.SslHandler - [id: 0xde05384c, L:/0:0:0:0:0:0:0:1:51695 - R:/0:0:0:0:0:0:0:1%lo:51409] HANDSHAKEN: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
17:06:51.804 [testsuite-oio-worker-6-1] DEBUG i.n.t.t.socket.SocketSslEchoTest - HANDSHAKEN:
	client { sent: 73317, rcvd: 0, nego: 1, cipher: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 },
	server { rcvd: 0, nego: 1, cipher: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 }
[CodeBlob (0x00007f1725051e10)]
Framesize: 0
BufferBlob (0x00007f1725051e10) used for StubRoutines (2)
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (sharedRuntime.cpp:834), pid=133636, tid=139737064572672
#  fatal error: exception happened outside interpreter, nmethods and vtable stubs at pc 0x00007f172505436f
#
# JRE version: Java(TM) SE Runtime Environment (8.0_40-b26) (build 1.8.0_40-b26)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.40-b25 mixed mode linux-amd64 compressed oops)
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/rrosenbl/work/netty/testsuite/hs_err_pid133636.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
/bin/sh: line 1: 133636 Aborted                 (core dumped) /export/apps/jdk/JDK-1_8_0_40/jre/bin/java -server -dsa -da -ea:io.netty... -XX:+AggressiveOpts -XX:+TieredCompilation -XX:+UseBiasedLocking -XX:+UseFastAccessorMethods -XX:+OptimizeStringConcat -XX:+HeapDumpOnOutOfMemoryError '-javaagent:/home/rrosenbl/.m2/repository/org/mortbay/jetty/alpn/jetty-alpn-agent/2.0.1/jetty-alpn-agent-2.0.1.jar=${jetty.alpnAgent.option}' -D_ -D_ -D_ -jar /home/rrosenbl/work/netty/testsuite/target/surefire/surefirebooter1581942141897288830.jar /home/rrosenbl/work/netty/testsuite/target/surefire/surefire2070898455139189286tmp /home/rrosenbl/work/netty/testsuite/target/surefire/surefire_04084957818531142856tmp
```
machine details:
```
$ uname -a
Linux rrosenbl-ld1 2.6.32-504.el6.x86_64 #1 SMP Tue Sep 16 01:56:35 EDT 2014 x86_64 x86_64 x86_64 GNU/Linux
$ java -version
java version "1.8.0_40"
Java(TM) SE Runtime Environment (build 1.8.0_40-b26)
Java HotSpot(TM) 64-Bit Server VM (build 25.40-b25, mixed mode)
$ mvn -version
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T08:41:47-08:00)
Maven home: /home/rrosenbl/apps/apache-maven-3.3.9
Java version: 1.8.0_40, vendor: Oracle Corporation
Java home: /export/apps/jdk/JDK-1_8_0_40/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "2.6.32-504.el6.x86_64", arch: "amd64", family: "unix"
```
the complete crash log is up at http://hastebin.com/raw/gilomisuci

**Comment by Scottmitch** [comment]
possible duplicate of https://github.com/netty/netty/issues/3256. Please update your JDK and report back if you can still reproduce.

**Comment by radai-rosenblatt** [comment]
indeed, with jdk 1.8.0_102-b14 everything builds fine...

**Comment by normanmaurer** [comment]
Thx for letting us know

**Comment by prakash-mannikeri** [comment]
jdk 1.8.0_102 - worked fine for me
